### PR TITLE
Specify tar file format explicitly

### PR DESCRIPTION
**Pull request by ahryniv**
In Python 3.8. [default format](https://docs.python.org/3/library/tarfile.html#tarfile.DEFAULT_FORMAT) of tarfiles was changed to `PAX_FORMAT`.

`dpkg` rejects deb packages that use tarfiles with PAX format with an "unsupported PAX tar header type" message.

Added optional parameter `--format` to set tar format explicitly. If it's not set `tarfile.DEFAULT_FORMAT` would be used.

see related:
* https://github.com/bazelbuild/rules_pkg/pull/217
* https://github.com/bazelbuild/rules_pkg/pull/601

### CODE REVIEW

File: tools/build_defs/pkg/archive.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def __init__(self,
     self.preserve_mtime = preserve_tar_mtimes
 
     self.fileobj = None
-    self.tar = tarfile.open(name=name, mode=mode, fileobj=self.fileobj)
+    self.tar = tarfile.open(name=name, mode=mode, fileobj=self.fileobj, format=tarfile.GNU_FORMAT)
```
Comment (cwoerner) [inline tools/build_defs/pkg/archive.py:50]: Rather than hacking archive.py, can pkg.archive.tar be monkey patched at runtime from within our code?

Comment (cwoerner) [reply tools/build_defs/pkg/archive.py:50]: Oh, I see, it's a context manager object, so likely no.   Huh, ok, let's discuss more.

Comment (cwoerner) [reply tools/build_defs/pkg/archive.py:50]: If you hope to push this upstream you shouldn't set format=tarfile.GNU_FORMAT in the tarfile.open method invocation, but rather define an additional named argument  to the TarFileWriter constructor called "format" with a default value (e.g format=tarfile.DEFAULT), then pass that into the tarfile.open() function.

**Review by cwoerner — CHANGES_REQUESTED** [review]

File: tools/build_defs/pkg/archive.py (hunk 1 above)
Comment (cwoerner) [reply tools/build_defs/pkg/archive.py:50]: Also, add tests to archive_test.py for the parameter change I suggested above.

**Review by aiuto — COMMENTED** [review]
We will not take this patch.  The internal pkg_tar is deprecated and slowly being removed. Users must migrate to https://github.com/bazelbuild/rules_pkg.

**Pull request closed without merging**